src/utils/updateStorage.ts

```diff
@@ -0,0 +1,57 @@
+import { UserSession } from "~types/userTypes"
+
+/**
+ *
+ * @description recursive function that will check every value of the existing storage against the provided values and update as needed
+ * @param currentValue the current state in storage
+ * @param newValue any values to be changed in our stored state
+ * @param index do not pass anything, it will automatically default to 0
+ * @param changes do not pass anything, it will default to an empty object
+ * @returns a new full UserSession object to pass into storage with the updated values
+ */
+export default function updateStorage(
+  currentValue: UserSession,
+  newValue: Partial<Omit<UserSession["user"], "audience_member">> &
+    Partial<UserSession["user"]["audience_member"]>,
+  index: number = 0,
+  changes = {}
```

Comment:
As per our conversation, this can be a globally scoped variable instead of a parameter

Reply:
Yes, I will [curry](https://mostly-adequate.gitbook.io/mostly-adequate-guide/ch04) it